gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -122,7 +119,7 @@ public Optional<Dag<JobExecutionPlan>> validateAndHandleConcurrentExecution(Conf
       sharedFlowMetricsSingleton.conditionallyUpdateFlowGaugeSpecState(spec,
           SharedFlowMetricsSingleton.CompiledState.SKIPPED);
       Instrumented.markMeter(sharedFlowMetricsSingleton.getSkippedFlowsMeter());
-      if (!isScheduledFlow((FlowSpec) spec)) {
+      if (!((FlowSpec) spec).isScheduled()) {
```

Comment:
Is there a scenario where `spec` isn't a flowspec here? Maybe it makes sense for the function to require a flowspec so we don't need to keep casting.

Reply:
Good callout, from checking the code we only call these functions with a FlowSpec and otherwise throw an error. Changing it to FlowSpec.